### PR TITLE
Commodore64x #1982 - Added module bans

### PR DESCRIPTION
**Pull request by ghost**
Issue reference: #1982
Added.

### CODE REVIEW

**Comment by mayokek** [comment]
Hmm, did you test this?

**Comment by ItzLevvie** [comment]
You're missing a '.java' extension on the file 'Module_bans'

**Comment by OxLemonxO** [comment]
You didn''t put .java at the end of the Module_bans class's filename

**Comment by ghost** [comment]
@OxLemonxO It compiles and there are no outstanding issues.

**Comment by ghost** [comment]
@mayokek Yes I did, it follows very closely the code for the permban module.

**Comment by ItzLevvie** [comment]
Also, your code needs to be formatted into Allman Style @Commodore64x

**Comment by ghost** [comment]
Done all changes

**Comment by ghost** [comment]
Anyways, you forgot to edit banmananager.java

Add this into banmanager.java

    public static final String CONFIG_FILENAME = "bans.yml";

File: src/main/java/me/totalfreedom/totalfreedommod/banning/BanManager.java [hunk 1]
```diff
@@ -30,13 +30,14 @@
     private final Map<String, Ban> ipBans = Maps.newHashMap();
     private final Map<String, Ban> nameBans = Maps.newHashMap();
     private final List<String> unbannableUsernames = Lists.newArrayList();
+    public static final String CONFIG_FILENAME = "bans.yml";
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/banning/BanManager.java:33]: This should be being pulled from BanManager.java so as to reduce the likelihood of typos.

File: src/main/java/me/totalfreedom/totalfreedommod/httpd/module/Module_bans.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            if (!isAuthorized(remoteAddress))
+            {
+                FLog.info("An unregistered IP (" + remoteAddress + ") has tried to access the ban list.");
+                return new NanoHTTPD.Response(NanoHTTPD.Response.Status.NOT_FOUND, NanoHTTPD.MIME_PLAINTEXT,
+                        "You may not view the banlist, Your IP, " + remoteAddress + ", is not registered to an admin on the server.");
+
+            }
+            else
+            {
+                return HTTPDaemon.serveFileBasic(new File(plugin.getDataFolder(), BanManager.CONFIG_FILENAME));
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/httpd/module/Module_bans.java:35]: This should never be able to be reached?

File: src/main/java/me/totalfreedom/totalfreedommod/httpd/module/Module_bans.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    {
+        File banFile = new File(plugin.getDataFolder(), BanManager.CONFIG_FILENAME);
+        if (banFile.exists())
+        {
+            final String remoteAddress = socket.getInetAddress().getHostAddress();
+            if (!isAuthorized(remoteAddress))
+            {
+                FLog.info("An unregistered IP (" + remoteAddress + ") has tried to access the ban list.");
+                return new NanoHTTPD.Response(NanoHTTPD.Response.Status.NOT_FOUND, NanoHTTPD.MIME_PLAINTEXT,
+                        "You may not view the banlist, Your IP, " + remoteAddress + ", is not registered to an admin on the server.");
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/httpd/module/Module_bans.java:30]: Also, this should be returning a 403 and not a 404 error.

Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/httpd/module/Module_bans.java:30]: Do we really need to feed back the players IP? Don't see that being of any use...

File: src/main/java/me/totalfreedom/totalfreedommod/httpd/module/Module_bans.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    @Override
+    public NanoHTTPD.Response getResponse()
+    {
+        File banFile = new File(plugin.getDataFolder(), BanManager.CONFIG_FILENAME);
+        if (banFile.exists())
+        {
+            final String remoteAddress = socket.getInetAddress().getHostAddress();
+            if (!isAuthorized(remoteAddress))
+            {
+                FLog.info("An unregistered IP (" + remoteAddress + ") has tried to access the ban list.");
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/httpd/module/Module_bans.java:28]: Should this be flagging as a warning?

**Review by Wild1145 — CHANGES_REQUESTED** [review]

**Comment by Wild1145** [comment]
Due to the original repo having been destroyed, I am closing this PR down as a result.

**Pull request closed without merging**